### DISCUSSION TITLE
Integrate NAS shares for LTSP clients

### DISCUSSION THREAD

**Discussion by sogeil** [ltsp/ltsp#834] [Q&A]
Hello!

On my current server (x2go-TCE = x2go + pxe + ThinClients) I have integrated Samba shares of a NAS in the /etc/fstab via cifs.
Example:
`//192.168.1.1/Diverses/		/home/gemeinsame_Daten/Diverses		cifs	credentials=/home/me/.NAS_credentials,uid=1000,gid=1012,file_mode=0775,dir_mode=0775		0	0`

I regulate the access rights of the users/groups via the folder rights of /home/gemeinsame_Daten/Diverses and others.

How do I achieve this or something similar with the LTSP clients?
I can't get a folder into /home or the folder I created on the server isn't visible on the client. Elsewhere it works, e.g. in /media . But only if I write the username and password directly in /etc/ltsp/ltsp.conf.
Reading a credentials file in my /home/me doesn't work at all.

How do you correctly integrate the shares of a NAS with an LTSP server?

Greeting,
Olaf

**Comment by alkisg** [comment]
You could easily do that by putting the credentials file in e.g. /var/lib/private/.NAS_credentials; then `ltsp image /` would put it in the LTSP image (while /home doesn't go in the image).
But then anyone with root access on the network could mount the LTSP image and read that file.
That's the main problem with diskless clients, it's difficult to store secrets in them. [This thread](https://github.com/ltsp/ltsp/issues/127#issuecomment-733511783) has some thoughts on how to have secrets in diskless clients; with UEFI and TPM it should be easier.

So long story short, secrets should be accessed after login, and such mounts should happen after login:

- A user logs in
- An /etc/xdg/autostart/shared-mount.desktop startup entry is launched
- Which runs a /usr/local/bin/shared-mount script
- Which runs something like:

      mkdir ~/shared-folder
      ltsp remoteapps cat /credentials/file/on/server >~/shared-folder/credentials
      gio mount smb://server... <~/shared-folder/credentials

See [this page](https://forum.manjaro.org/t/root-tip-utility-script-gio-mount-samba-share/100723) for how to use gio mount with a credentials file.

Btw, this would be a lot easier if Bodhi has a keyring where the users can store the credentials for samba mounts, then you don't need to manually run gio. Unfortunately for gnome, the gnome-keyring daemon doesn't work over SSHFS.

**Comment by sogeil** [comment]
Hello!

Thanks a lot, I solved it with the first method, works now after a few failures.

Greeting,
Olaf
